packages/cardano-services/test/util/TypeormProvider.test.ts

```diff
@@ -0,0 +1,43 @@
+import { BehaviorSubject, of } from 'rxjs';
+import { TypeormProvider } from '../../src/util';
+import { getEntities } from '../../src';
+import { logger } from '@cardano-sdk/util-dev';
+import { projectorConnectionConfig } from '../util';
+
+class TestProvider extends TypeormProvider {}
+
+jest.mock('../../src/util/createTypeormDataSource', () => ({
+  createTypeormDataSource: jest.fn().mockImplementation(() => {
+    let called = false;
+
+    return of({
+      query: () => {
+        if (called) return Promise.reject(0);
+
+        called = true;
+        return Promise.resolve(0);
+      }
+    });
+  })
+}));
+
+describe('TypeormProvider', () => {
+  const connectionConfig$ = new BehaviorSubject(projectorConnectionConfig);
+  const entities = getEntities(['currentPoolMetrics', 'poolMetadata', 'poolDelisted']);
+  const provider = new TestProvider('test', { connectionConfig$, entities, logger });
+
+  beforeAll(async () => {
+    await provider.initialize();
+    await provider.start();
+  });
+
+  afterAll(() => provider.shutdown());
+
+  test('healthCheck', async () => {
+    expect((await provider.healthCheck()).ok).toBeTruthy();
+    expect((await provider.healthCheck()).ok).toBeFalsy();
+    expect((await provider.healthCheck()).ok).toBeTruthy();
```

Comment:
Took a while to understand how this works, would be nice to add a comment
```suggestion
    expect((await provider.healthCheck()).ok).toBeTruthy();
    expect((await provider.healthCheck()).ok).toBeFalsy();
    // data source is re-created
    expect((await provider.healthCheck()).ok).toBeTruthy();
```